scripts/scil_compute_qbx.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import argparse
+from operator import itemgetter
+import os
+
+from dipy.io.stateful_tractogram import Space, StatefulTractogram
+from dipy.io.streamline import save_tractogram
+from dipy.segment.clustering import qbx_and_merge
+
+from scilpy.io.streamlines import load_tractogram_with_reference
+from scilpy.io.utils import (add_overwrite_arg,
+                             add_reference_arg,
+                             assert_inputs_exist,
+                             assert_outputs_exist,
+                             assert_output_dirs_exist_and_empty)
+
+DESCRIPTION = """
+    Compute clusters using QuickBundlesX and save them separately.
+    We cannot know the number of clusters in advance.
+"""
+
+
+def _build_args_parser():
+    p = argparse.ArgumentParser(description=DESCRIPTION,
+                                formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_tractogram',
+                   help='Tractogram filename.\n'
+                   'Path of the input tractogram or bundle.')
+    p.add_argument('dist_thresh', type=float,
+                   help='Last QuickBundlesX threshold in mm. Typically \n'
+                   'the value are between 10-20mm.')
+    p.add_argument('output_clusters_dir',
+                   help='Path to the clusters directory.')
+
+    p.add_argument('--nb_points', type=int, default='20',
+                   help='Streamlines will be resampled to have this '
+                        'number of points [%(default)s].')
+    p.add_argument('--output_centroids',
+                   help='Output tractogram filename.\n'
+                   'Format must be readable by the Nibabel API.')
+
+    add_reference_arg(p)
+    add_overwrite_arg(p)
+
+    return p
+
+
+def main():
+    parser = _build_args_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, args.in_tractogram)
+    assert_outputs_exist(parser, args, [], optional=args.output_centroids)
+    if args.output_clusters_dir:
```

Comment:
Unneeded if. output_clusters_dir is positional, so it is required to be supplied when calling the script